### PR TITLE
ACI: New aci_maintenance_policy module

### PR DESCRIPTION
**Pull request by sgerhart**
##### SUMMARY

Creates a firmware maintenance policy to be used with a maintenance group

##### ISSUE TYPE

- New Module Pull Request

##### COMPONENT NAME

aci_maintenance_policy
##### ANSIBLE VERSION

```paste below
ansible 2.8.0.dev0 (maintenance_policy_module e724dcf66b) last updated 2018/11/08 14:32:36 (GMT -400)
  config file = None
  configured module search path = [u'/Users/stgerhar/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/stgerhar/Development/Python/github.com/sgerhart/ansible-dev/lib/ansible
  executable location = /Users/stgerhar/Development/Python/github.com/sgerhart/ansible-dev/bin/ansible
  python version = 2.7.14 (default, Mar 10 2018, 00:01:04) [GCC 4.2.1 Compatible Apple LLVM 9.0.0 (clang-900.0.39.2)]
```

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

**Comment by dagwieers** [comment]
cc @rsmeyers

**Comment by rsmeyers** [comment]
Can you adminst triggered so we can trigger the maintenance policy once attached to a maintenance group ?

See also

method: POST
url: https://10.48.22.44/api/node/mo/uni/fabric.json
payload{"maintMaintP":{"attributes":{"dn":"uni/fabric/maintpol-EvenMaintGrp","adminSt":"triggered"},"children":[]}}
response: {"totalCount":"0","imdata":[]}

**Comment by sgerhart** [comment]
@rsmeyers - I guess that would be a possibility but I just used the same settings that would be created by the APIC directly. The adminSt is "untriggered". What would the result of setting the adminSt setting triggered? If it would prevent the ability to use a policy for multiple groups, then that would be ideal.

**Comment by rsmeyers** [comment]
When you set adminst to triggered, we will actually perform the upgrade, else, it's just a policy doing nothing....

**Comment by sgerhart** [comment]
That makes sense, so I will add it to the module.

File: lib/ansible/modules/network/aci/aci_maintenance_policy.py [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+
+from ansible.module_utils.network.aci.aci import ACIModule, aci_argument_spec
+from ansible.module_utils.basic import AnsibleModule
+
+
+def main():
+    argument_spec = aci_argument_spec()
+    argument_spec.update(
+        name=dict(type='str', aliases=['maintenance_policy']),  # Not required for querying all objects
+        runmode=dict(type='bool', default=True),
```
Comment (dagwieers) [inline lib/ansible/modules/network/aci/aci_maintenance_policy.py]: I wonder if this should be a boolean, or a list of options. My worry is that in the future this gets extended to more than two options, and we're stuck with having it as a boolean.

Comment (dagwieers) [reply lib/ansible/modules/network/aci/aci_maintenance_policy.py]: Also defaulting to `True` will always override the value, even if it was not specified.
This is probably not what we want here.

File: lib/ansible/modules/network/aci/aci_maintenance_policy.py [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+  graceful:
+    description:
+    - Whether the system will bring down the nodes gracefully during an upgrade, which reduces traffic lost.
+    type: bool
+  scheduler:
+    description:
+    - The name of scheduler that is applied to the policy.
+    type: str
+    required: true
+  upgrade:
```
Comment (dagwieers) [inline lib/ansible/modules/network/aci/aci_maintenance_policy.py]: To me it is unclear why it is called `upgrade` here, but the description mentions `adminst`.

**Review by dagwieers — CHANGES_REQUESTED** [review]
A few remarks

**Review by dagwieers — APPROVED** [review]
LGTM

**Pull request merged**